rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,148 @@
+# RFC: General File Naming Guidelines - Convergence
+
+<!--
+An RFC can be anything. A question, a suggestion, a plan. The purpose of this template is to give some structure to help folks write successful RFCs. However, don't feel constrained by this template; use your best judgement.
+
+Tips for writing a successful RFC:
+
+- Simple plain words that make your point, fancy words obfuscate
+- Try to stay concise, but don't gloss over important details
+- Try to write a neutral problem statement, not one that motivates your desired solution
+- Remember, "Writing is thinking". It's natural to realize new ideas while writing your proposal
+-->
+
+---
+
+_List contributors to the proposal : @hotell_
+
+## Summary
+
+<!-- Explain the proposed change -->
+
+Apply consistent, easy to navigate style guide for naming files within converged components.
+
+## Background
+
+<!-- If there is relevant background include it here -->
+
+This RFC builds on [RFC for convergence implementation patterns](https://github.com/microsoft/fluentui/pull/16806/files).
+
+## Problem statement
+
+<!--
+Why are we making this change? What problem are we solving? What do we expect to gain from this?
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+Currently converged package contains following files:
+
+> We'll use contrived name of package - `example` which lives in `/packages/example` folder
+
+- `index.ts` - barrel / Public API surface
+- `Example.tsx` - (complete component to be used as is)
+- `useExample.ts` - (CONTROLLER/MODEL /main logic behind component - re-usable hook)
+- `useExampleClasses.ts` - (STYLES/ main logic for styling component - re-usable hook)
+- `useExampleRandom.tsx` - (some encapsulated logic - not really a hook)
```

Comment:
Hmm, could we then maybe put this inside a `useExample` dir to signify that it's additional encapsulated logic? Would that make sense?

Reply:
Oh wait, this is what we currently have. Nvm.